Eradicating url() calls

arildm opened the pull request.

arildm committed:
Eradicating url() calls

arildm committed:
Added missing empty url() params

larowlan commented:
looks like dupe of #39?

arildm commented:
They are related but contain different changes. Didn't see #39 before creating this.

larowlan commented:
ok, will take a look asap

On 16 October 2014 16:39, Arild notifications@github.com wrote:

> They are related but contain different changes. Didn't see #39
> https://github.com/larowlan/token/pull/39 before creating this.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/larowlan/token/pull/41#issuecomment-59320164.

## 

Lee Rowlands
Senior Drupal Developer
PreviousNext
skype: larowlan
email: larowlan@previousnext.com.au

Berdir commented on src/Tests/TokenCommentTestCase.php:
```diff
@@ -35,9 +35,9 @@ function testCommentTokens() {
-      'url' => url('comment/' . $parent_comment->id(), array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => TRUE)),
-      'url:absolute' => url('comment/' . $parent_comment->id(), array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => TRUE)),
-      'url:relative' => url('comment/' . $parent_comment->id(), array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => FALSE)),
+      'url' => \Drupal::url('entity.comment.canonical', ['comment' => $parent_comment->id()], array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => TRUE)),
```
I think comments have a permalink() method that should return the fragment part?

Berdir commented on src/Tests/TokenCurrentPageTestCase.php:
```diff
@@ -40,13 +40,13 @@ function testCurrentPageTokens() {
-      '[current-page:url]' => url("node/{$node->id()}", array('absolute' => TRUE)),
-      '[current-page:url:absolute]' => url("node/{$node->id()}", array('absolute' => TRUE)),
-      '[current-page:url:relative]' => url("node/{$node->id()}", array('absolute' => FALSE)),
+      '[current-page:url]' => \Drupal::url('entity.node.canonical', ['node' => $node->id()], array('absolute' => TRUE)),
```
I suggest you use $node->url('canonical', array('absolute' => TRUE) and so on for this.

Berdir commented on token.tokens.inc:
```diff
@@ -479,7 +480,7 @@ function token_tokens($type, array $tokens, array $data = array(), array $option
           break;
         case 'edit-url':
-          $replacements[$original] = url("admin/structure/types/manage/{$node_type->type}", $url_options);
+          $replacements[$original] = \Drupal::url('entity.node_type.edit_form', ['node_type' => $node_type->type], $url_options);
```
Also here, $node_type->url('edit-form') should give you the same.

Berdir commented on token.tokens.inc:
```diff
@@ -808,8 +809,8 @@ function token_tokens($type, array $tokens, array $data = array(), array $option
-          if (_token_module($type, 'url') == 'token' && $uri = $entity->urlInfo()->toArray()) {
-            $replacements[$original] = url($uri['path'], $uri['options']);
+          if (_token_module($type, 'url') == 'token' && $url = $entity->urlInfo()) {
+            $replacements[$original] = $url;
```
This is not correct, should probably use url() instead, the token replacement needs to be a string.

Berdir replied:
Same for the nested url tokens below. The current url type expects route name, this passes along url apparently I think type url should expect a Url object, not route_name and so on, like most other API's in 8.x now. So this should pass along array('url' => $entity->urlInfo()) instead and the logic for type url needs to be reworked then.

Commit:
Modifications done according to Sascha's comments

Commit:
used the url() for the nested url tokens as well

arildm commented:
Some tests are failing but we couldn't figure out if they were introduced with these changes or not. We should probably run the tests before these changes, with core branch rolled back to before url() was removed, in order to compare test results.

arildm committed:
Fixed TokenCurrentPageTestCase

arildm committed:
Fixed TokenCommentTestCase

Could not use `permalink()` because it returns `node/1#comment-1` when we
expect `comment/1#comment-1`. We _should_ probably change that expectation, but
that requires some further tinkering, e.g. the part in `token_tokens()` that
renders `[(any entity):url]` calls `$entity->url()` which differs from
`$comment->permalink()`.

arildm committed:
Fixed some of TokenURLTestCase and TokenMenuTestCase

TokenMenuTestCase needs menu_link_content-related changes but not in this pull
request. I did sneak in a YAML 'entity:' key update though.

arildm commented:
Yay, the last commits pass the tests. Some menu link-related fails remain but they are another issue. Please review, and sorry for the mess!
